src/compiler/scala/tools/nsc/typechecker/RefChecks.scala

```diff
@@ -1208,10 +1200,38 @@ abstract class RefChecks extends InfoTransform with scala.reflect.internal.trans
           // trait T { def f: Object }; object O extends T { object f }. Need to generate method f in O.
           if (module.isOverridingSymbol) matchingInnerObject() else Nil
         else
-          newInnerObject()
+          newInnerObject(site, module)
       )
       transformTrees(newTrees map localTyper.typedPos(moduleDef.pos))
     }
+    def newInnerObject(site: Symbol, module: Symbol): List[Tree] = {
+      // Create the module var unless it is already in the module owner's scope.
```

Comment:
Nope, I should have removed the comment as part of resolving the merge conflicts after rebasing on your changes to this area.